content/ch-gates/oracles.ipynb

```diff
@@ -0,0 +1,289 @@
+      "text/plain": [
+       "<Figure size 423.808x204.68 with 1 Axes>"
+      ]
+     },
+     "execution_count": 25,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "qc = Uf + Vf.inverse()\n",
+    "\n",
+    "qc.draw(output='mpl',justify='none')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This circuit begins with two identical `cx` gates, whose effects cancel each other out. All that remains is the final `cx` between the input and garbage registers. Mathematically, this means\n",
+    "\n",
+    "$$\n",
+    "V_f^\\dagger U_f \\left| x,f(x),0 \\right\\rangle = \\left| x , 0 ,g(x) \\right\\rangle.\n",
+    "$$\n",
+    "\n",
+    "This does not leave us with the state we require, but instead leaves the first qubit entangled with unwanted garbage. Any subsequent steps in an algorithm will therefore not run as expected, since the state is not the one that we need.\n",
+    "\n",
+    "For this reason we need a way of removing classical garbage from our quantum algorithms. This can be done by a method known as 'uncomputation. We simply need to take another blank variable and apply $V_f$\n",
+    "\n",
+    "\n",
+    "$$\n",
+    "\\left| x, 0, 0, 0 \\right\\rangle \\rightarrow \\left| x,f(x),g(x),0 \\right\\rangle.\n",
+    "$$\n",
+    "\n",
+    "Then we apply a set of controlled-NOT gates, each controlled on one of the qubits used to encode the output, and targeted on the corresponding quit in the extra blank variable. This effectively copyies the information over.\n",
```

Comment:
- targeted on the corresponding quit in the extra blank variable : quit -> qubit

- I think if the sentence, 
"Then we apply a set of controlled-NOT gates, each controlled on one of the qubits used to encode the output, and targeted on the corresponding quit in the extra blank variable."
include the fact that there is only one output qubit and one extra blank variable in the example, it would be helpful to the reader.  just another personal opinion, if I understand the content right. :)